static/app/views/settings/organizationTeams/teamDetails.tsx

```diff
@@ -96,119 +52,115 @@ class TeamDetails extends React.Component<Props, State> {
+                            size="small"
+                            onClick={() => handleRequestAccess(team)}
+                          >
+                            {team.isPending ? t('Request Pending') : t('Request Access')}
+                          </Button>
+                        </RequestAccessWrapper>
+                      ) : (
+                        <div>{t('You do not have access to this team.')}</div>
+                      )}
+                    </Alert>
+                  );
+                }
+                return (
+                  <div key={i}>
+                    <SentryDocumentTitle
+                      title={t('Team Details')}
+                      orgSlug={props.params.orgId}
+                    />
+                    <h3>
+                      <IdBadge hideAvatar team={team} avatarSize={36} />
+                    </h3>
+
+                    <NavTabs underlined>{navigationTabs}</NavTabs>
+
+                    {isValidElement(children) &&
+                      cloneElement(children, {
+                        team,
+                        onTeamChange: () => onTeamChange(team),
+                      })}
+                  </div>
+                );
+              })
+            ) : (
+              <Alert type="warning">
+                <div>{t('You do not have access to this team.')}</div>
+              </Alert>
+            )
           ) : (
-            <div>{t('You do not have access to this team.')}</div>
+            <LoadingIndicator />
           )}
-        </Alert>
-      );
-    }
-    if (error) {
-      return <LoadingError onRetry={this.fetchData} />;
-    }
-
-    // `/organizations/${orgId}/teams/${teamId}`;
-    const routePrefix = recreateRoute('', {routes, params, stepBack: -1});
-
-    const navigationTabs = [
-      <ListLink key={0} to={`${routePrefix}members/`}>
-        {t('Members')}
-      </ListLink>,
-      <ListLink key={1} to={`${routePrefix}projects/`}>
-        {t('Projects')}
-      </ListLink>,
-      <ListLink key={2} to={`${routePrefix}notifications/`}>
-        {t('Notifications')}
-      </ListLink>,
-      <ListLink key={3} to={`${routePrefix}settings/`}>
-        {t('Settings')}
-      </ListLink>,
-    ];
-
-    return (
-      <div>
-        <SentryDocumentTitle title={t('Team Details')} orgSlug={params.orgId} />
-        <h3>
-          <IdBadge hideAvatar team={team} avatarSize={36} />
-        </h3>
-
-        <NavTabs underlined>{navigationTabs}</NavTabs>
-
-        {React.isValidElement(children) &&
-          React.cloneElement(children, {
-            team,
-            onTeamChange: this.onTeamChange,
-          })}
-      </div>
-    );
-  }
+        </div>
+      )}
+    </Teams>
+  );
 }
 
-// TODO(davidenwang): change to functional component and replace withTeams with useTeams
-export default withApi(withOrganization(withTeams(TeamDetails)));
+export default withApi(withOrganization(TeamDetails));
```

Comment:
you're killing it, replace withOrganization with useOrganization and withApi with useApi